src/CoreWCF.Primitives/src/CoreWCF/Security/Authorization/AuthorizationServiceBehavior.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Reflection;
+using CoreWCF.Channels;
+using CoreWCF.Description;
+using Microsoft.AspNetCore.Authorization;
+
+namespace CoreWCF.Security.Authorization
+{
+    public class AuthorizationServiceBehavior : IServiceBehavior
+    {
+        private readonly IAuthorizationPolicyProvider _authorizationPolicyProvider;
+        private readonly IAuthorizationService _authorizationService;
+
+        public AuthorizationServiceBehavior(IAuthorizationPolicyProvider authorizationPolicyProvider,
+            IAuthorizationService authorizationService)
+        {
+            _authorizationPolicyProvider = authorizationPolicyProvider;
+            _authorizationService = authorizationService;
+        }
+
+        public void Validate(ServiceDescription serviceDescription, ServiceHostBase serviceHostBase)
+        {
+
+        }
+
+        public void AddBindingParameters(ServiceDescription serviceDescription, ServiceHostBase serviceHostBase,
+            Collection<ServiceEndpoint> endpoints,
+            BindingParameterCollection bindingParameters)
+        {
+
+        }
+
+        public void ApplyDispatchBehavior(ServiceDescription serviceDescription, ServiceHostBase serviceHostBase)
+        {
+            foreach (var endpoint in serviceDescription.Endpoints)
+            {
+                foreach (var operation in endpoint.Contract.Operations)
+                {
+                    MethodInfo methodInfo = GetServiceImplMethodInfo(serviceDescription, operation);
+                    bool hasAllowAnonymous = methodInfo.GetCustomAttribute(typeof(AllowAnonymousAttribute)) is not null;
+                    var authorizeData = methodInfo
+                        .GetCustomAttributes(typeof(AuthorizeAttribute), false).OfType<IAuthorizeData>()
+                        .ToArray();
+
+                    operation.OperationBehaviors.Add(
+                        new AuthorizationOperationBehavior(authorizeData, _authorizationPolicyProvider,
```

Comment:
Rather than adding a new operation behavior, I think we should also add a property on OperationDescription to hold the IAuthorizeData list. The list would be copied across in DispatcherBuilder.BuildDispatchOperation. A reasonable small extra cost is okay on startup for an unused feature, as long as there isn't an unnecessary cost at runtime. We should populate this data in TypeLoader.CreateOperationDescription. Doing it in a behavior means any reflected data may or may not be there before a users behavior executes as it would be order dependent.

Reply:
Thinking about this more as I've explored the code, I think OperationDescription should have a collection of IAuthorizeData and DispatchOperation should store an AuthorizePolicy. OperationDescription holds object models to describe the operation whereas DispatchOperation holds concrete types with functionality so this seems like the right way to do things.

Reply:
`OperationDescription` now holds a `Collection<IAuthorizeData>` named `AuthorizeData`.
`DispatchOperation` now hols the `AuthorizationPolicy` in a property named `AuthorizationPolicy`.